packages/angular_devkit/build_angular/src/builders/ssr-dev-server/index.ts

```diff
@@ -376,23 +376,22 @@ function getSslConfig(
 }
 
 async function getProxyConfig(root: string, proxyConfig: string): Promise<MiddlewareHandler[]> {
-  const proxy = await loadProxyConfiguration(root, proxyConfig, false);
+  const proxy = await loadProxyConfiguration(root, proxyConfig, true);
   const createdProxies = [];
   const { createProxyMiddleware } = await import('http-proxy-middleware');
+
   for (const [key, context] of Object.entries(proxy)) {
-    if (typeof key === 'string') {
+    if (key.startsWith('^') && key.endsWith('$')) {
```

Comment:
This seems like it is a breaking change, which is not desired. 

In the below code, the proxy key is modified to behave the same way as webpack-dev-server seeing that this builder is intended to be used with webpack based builders.

Example, with webpack-dev-server,  `/api/*` incorrectly matches `/api/foo/bar` whilst with vite it will not.

Reply:
I have created a function with which the generated regex is adapted as before.
The test has also been adapted accordingly to check the behaviour with more than one path segment.